api/buckets/service.go

```diff
@@ -2131,18 +2213,50 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 				return nil, fmt.Errorf("there is an in progress archive")
 			// Case 1.c.
 			case ffs.Failed, ffs.Canceled:
-				jid, err = s.PGClient.FFS.PushStorageConfig(ctxFFS, p.Cid(), powc.WithOverride(true))
+				jid, err = s.PGClient.FFS.PushStorageConfig(ctxFFS, p.Cid(), powc.WithStorageConfig(storageConfig), powc.WithOverride(true))
 				if err != nil {
 					return nil, fmt.Errorf("pushing config: %s", err)
 				}
 			default:
 				return nil, fmt.Errorf("unexpected current archive status: %d", ba.Archives.Current.JobStatus)
 			}
 		} else { // Case 2.
-			jid, err = s.PGClient.FFS.Replace(ctxFFS, oldCid, p.Cid())
+			res, err := s.PGClient.FFS.GetStorageConfig(ctxFFS, oldCid)
 			if err != nil {
-				return nil, fmt.Errorf("replacing cid: %s", err)
+				return nil, fmt.Errorf("looking up old storage config: %s", err)
+			}
+			// ToDo: Remove this once .PGClient.FFS.GetStorageConfig returns a ffs.StorageConfig
+			var oldStorageConfig *ffs.StorageConfig
+			if res.Config != nil {
+				oldStorageConfig = &ffs.StorageConfig{
+					Repairable: res.Config.Repairable,
+					Cold:       fromRPCColdConfig(res.Config.Cold),
+					Hot:        fromRPCHotConfig(res.Config.Hot),
+				}
+			}
+
+			if reflect.DeepEqual(&storageConfig, oldStorageConfig) {
+				// Old storage config is the same as the new so use replace.
+				jid, err = s.PGClient.FFS.Replace(ctxFFS, oldCid, p.Cid())
+				if err != nil {
+					return nil, fmt.Errorf("replacing cid: %s", err)
+				}
+			} else {
+				// New storage config, so remove and push.
```

Comment:
I think that for the end state that you want to exist the answer is: yes, we are on the same page about that. Hot storage will be disabled.

Just to clear some details up, in the scenario you describe, `oldStorageConfig` won't be `nil`, it will be whatever the previous bucket archive used, probably has hot storage enabled, etc. The new `storageConfig` will likely be different because it must have hot storage disabled, and likely has other differences because of the new default settings. So for that reason, we will `Remove` and `Push` the new storage config with hot storage disabled.

Reply:
👍🏼